mitodl/micromasters
# Added field to FinalGrade to save if user paid on edx

#### What are the relevant tickets?
closes #2507 

#### What's this PR do?
Adds the feature to freeze the payment status if done on edX. This is important because we will not need to rely on cached enrollments to determine if a user payed on edX.

#### How should this be manually tested?
set `FEATURE_FINAL_GRADE_ALGORITHM=v1` in your `.env`

For the dashboard: 

in a not Financial Aid program, take a course (**that run in the past**) without an enrollment and create a final grade (using the factory) paying attention to set the `passed=True` and `course_run_payed_on_edx=True`. The course should show up in the dashboard as passed. 
Modifying the `course_run_payed_on_edx=False` should turn the course to "missed deadline". 

For the grades api:

take a course in a Financial aid program, create a certificate with `certificate_type="honor"` and from a shell run `grades.api._compute_grade_for_fa`: this should return an object saying that the course has been payed on edX.
if you modify the `certificate_type="audit"` and run the same function, the returned object should say that the course has not been payed on edX.

#### Where should the reviewer start?
`dashboard/utils.py`
`grades/api.py`

#### Any background context you want to provide?
This should handle the case of old courses on edX where used that got `honor` enrollment and certificate should be considered as payed.
